### DISCUSSION TITLE
.NET version required

### DISCUSSION THREAD

**Discussion by Gitoffthelawn** [Hofknecht/SystemTrayMenu#373] [Q&A] [answered]
I want to verify that this tool with only work on Windows computers with .NET 6 installed, and will ***not*** work on Windows computers that only have prior versions of .NET installed.

**Comment by Hofknecht** [comment, marked as answer]
Hey @Gitoffthelawn 
Thank you for the feedback and the discussion.

Since the beginning of STM there has been a long process of frameworks. We started in the net4.5 up to net4.8.
We then had to switch to .net core 2, then .net core 3.1 to get into the microsoft store. The last step was to switch to the framework net5 and then net6 in order to always continue the program at the cutting edge of technology. Going back would be extremely time-consuming. It would be a parallel development. Even if we got donations for it, it probably wouldn't be worth the effort, it would be too much work, to develop and test foreach framework paralell. 

When I change the <TargetFramework>net6.0-windows10.0.22000.0</TargetFramework> to <TargetFramework>net48</TargetFramework>, then currently I get 406 errors and 97 warnings when compiling. 
Most of it could probably be solved with a lot of effort, e.g. that we are currently using C# 9.0 features and that they would have to be rolled back to C#7.3. 

Of course, we understand that many people don't want to install an additional framework or that certain rights are often necessary in companies. There is a discussion here about whether the new frameworks should be pre-installed in new windows versions:
https://github.com/dotnet/core/issues/6411
